File: day01/b.ts
const words = [
  "zero",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
];

const reverse = (x: string) => x.split("").reverse().join("");

const regex = new RegExp(`\\d|${words.join("|")}`);
const reverseRegex = new RegExp(
  `\\d|${(words.map(reverse)).join("|")}`,
);
const res = (await Deno.readTextFile("./input.txt"))
  .trim()
  .split("\n")
  .map((line: string) => {
    const a = line.match(regex)?.at(0)!;
    const b = reverse(reverse(line).match(reverseRegex)?.at(0)!);
    return [a, b].map((n) => n.length > 1 ? `${words.indexOf(n)}` : n).join("");
  })
  .reduce((a: number, c: string) => a + parseInt(c), 0);
console.log(res);

File: day01/b.exs
valid_numbers = [
  "zero",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine"
]

regex = ~r/(#{Enum.join(valid_numbers, "|")}|\d)/

File.read!("input.txt") 
|> String.split("\n", trim: true)
|> Enum.map(fn l ->
    l
    |> then(&Regex.scan(regex, &1))
    |> Enum.map(&Enum.at(&1, 0))
    |> Enum.map(fn n ->
      n
      |> then(&Regex.match?(~r/\d/, &1))
      |> if(do: n, else: valid_numbers |> Enum.find_index(&(&1 == n)))
      end)
    |> then(&"#{Enum.at(&1, 0)}#{Enum.at(&1, -1)}")
    |> String.to_integer()
    end)
  |> Enum.sum()
|> IO.inspect()
